Source/WebKit/WebProcess/com.apple.WebProcess.sb.in

```diff
@@ -1208,33 +1205,27 @@
 
 #if PLATFORM(MAC)
 (define (notifyd-message-numbers) (message-number 1002 1010 1011 1012 1016 1017 1018 1021 1022 1025 1026 1028 1029 1030 1031 1032))
-
-;; FIXME should be removed when <rdar://problem/9347205> + related radar in Safari is fixed
-(if (equal? (param "ENABLE_SANDBOX_MESSAGE_FILTER") "NO")
-    (allow mach-lookup (global-name "com.apple.system.notification_center"))
-;; else
+(define (allow-notifyd)
+    (allow ipc-posix-shm-read* (ipc-posix-name "apple.shm.notification_center"))
     (allow mach-lookup (global-name "com.apple.system.notification_center")
         (apply-message-filter
             (deny mach-message-send)
             (deny mach-message-send (with no-report) (message-number 1023))
             (allow mach-message-send (notifyd-message-numbers)))))
 
-#if ENABLE(NOTIFY_BLOCKING)
-(if (equal? (param "ENABLE_SANDBOX_MESSAGE_FILTER") "YES")
-    (with-filter (state-flag "EnableExperimentalSandbox")
-        (allow mach-lookup (global-name "com.apple.system.notification_center")
-            (apply-message-filter
-                (deny mach-message-send)
-                (deny mach-message-send (with no-report) (message-number 1023))
-                (deny mach-message-send (with telemetry-backtrace) (notifyd-message-numbers))))))
+;; FIXME should be removed when <rdar://problem/9347205> + related radar in Safari is fixed
+(if (equal? (param "ENABLE_SANDBOX_MESSAGE_FILTER") "NO")
+    (allow mach-lookup (global-name "com.apple.system.notification_center"))
+;; else
+    (allow-notifyd))
```

Comment:
Can we merge the next rules here, something like:
```
if ENABLE(NOTIFY_BLOCKING)
    (with-filter (require-not (notify-blocking))
         (allow-notifyd)))
#else
    (allow-notifyd))
#endif
```
Then we don't need the deny rules?

Reply:
Good point, will fix!